cnd/src/http_api/routes.rs

```diff
@@ -335,71 +298,97 @@ struct SwapResource {
     pub role: Http<Role>,
 }
 
-trait GetSwapStatus {
-    fn get_swap_status(&self) -> SwapStatus;
-}
-
 trait GetAlphaEvents {
-    fn get_alpha_events(&self) -> LedgerEvents;
+    fn get_alpha_events(&self) -> Option<LedgerEvents>;
 }
 
 trait GetBetaEvents {
-    fn get_beta_events(&self) -> LedgerEvents;
+    fn get_beta_events(&self) -> Option<LedgerEvents>;
 }
 
 trait GetRole {
     fn get_role(&self) -> Role;
 }
 
-/// Return the ether swap quantity in wei.
-trait QuantityWei {
-    fn quantity_wei(&self) -> String;
-}
-
-/// Return the bitcoin swap quantity in satoshi.
-trait QuantitySatoshi {
-    fn quantity_satoshi(&self) -> String;
+trait GetAlphaParams {
+    type Output: Serialize;
+    fn get_alpha_params(&self) -> Self::Output;
 }
 
-fn herc20_halight_swap_status(
-    herc20_state: &herc20::State,
-    halight_state: &halight::State,
-) -> SwapStatus {
-    match (herc20_state, halight_state) {
-        (herc20::State::Redeemed { .. }, halight::State::Settled(_)) => SwapStatus::Swapped,
-        (herc20::State::IncorrectlyFunded { .. }, _) => SwapStatus::NotSwapped,
-        (herc20::State::Refunded { .. }, _) => SwapStatus::NotSwapped,
-        (_, halight::State::Cancelled(_)) => SwapStatus::NotSwapped,
-        _ => SwapStatus::InProgress,
-    }
+trait GetBetaParams {
+    type Output: Serialize;
+    fn get_beta_params(&self) -> Self::Output;
 }
 
+#[allow(clippy::large_enum_variant)]
 #[derive(Clone, Debug)]
-pub struct AliceHerc20HalightBitcoinSwap {
-    pub alpha_ledger_state: herc20::State,
-    pub beta_ledger_state: halight::State,
-
-    pub herc20_params: herc20::InProgressSwap,
-    pub halight_params: halight::InProgressSwap,
+pub enum AliceHerc20HalightBitcoinSwap {
+    Created {
+        // FIXME: this should be herc20::Asset
```

Comment:
Does it have to be? I think `herc20::Asset` is particularly useful if you want to express it in a single type. Within this big one, it just seems to be unnecessary wrapping as we are already in the context of `herc20`? But maybe consistency is better :)

Given that we are going to remove `DisplaySwap`, do we even still need `herc20::Asset`? :D

Reply:
These are all valid questions! We were definitely targetting consistency with this. I am leaning towards removing them altogether for now.

Reply:
I'm totally OK with removing `herc20::Asset` again.